src/test/pybind/test_rados.py

```diff
@@ -671,6 +681,23 @@ def cb(blah):
         eq(contents, b"bar")
         [i.remove() for i in self.ioctx.list_objects()]
 
+    def test_aio_writesame(self):
+        lock = threading.Condition()
+        count = [0]
+        def cb(blah):
+            with lock:
+                count[0] += 1
+                lock.notify()
+            return 0
+        comp = self.ioctx.aio_writesame("abc", b"rzx", 9, 0, cb, cb)
```

Comment:
the parameters passed here are inconsistent with the parameters in the function definition, can you run the tests and paste the output?

Reply:
updated:
test code:
```
from rados import (Rados, WriteOpCtx, ObjectNotFound)
from nose.tools import eq_ as eq, ok_ as ok, assert_raises
import threading
import rados
import unittest

class TestIoctx(unittest.TestCase):
    def setUp(self):
        self.rados = Rados(conffile='./ceph.conf')
        self.rados.connect()
        self.rados.create_pool('test_pol2')
        assert self.rados.pool_exists('test_pol2')
        self.ioctx = self.rados.open_ioctx('test_pol2')
    def tearDown(self):
        self.ioctx.close()
        self.rados.delete_pool('test_pol2')
        self.rados.shutdown()

    def test_writesame_op(self):
        with WriteOpCtx() as write_op:
            write_op.writesame(b'rzx', 9)
            self.ioctx.operate_write_op(write_op, 'abc')
            eq(self.ioctx.read('abc'), b'rzxrzxrzx')

    def test_writesame(self):
        self.ioctx.writesame('ob', b'rzx', 9)
        eq(self.ioctx.read('ob'), b'rzxrzxrzx')

    def test_aio_writesame(self):
        lock = threading.Condition()
        count = [0]
        def cb(blah):
            with lock:
                count[0] += 1
                lock.notify()
            return 0
        comp = self.ioctx.aio_writesame("abc", b"rzx", 9, 0, cb)
        comp.wait_for_complete()
        with lock:
            while count[0] < 1:
                lock.wait()
        eq(comp.get_return_value(), 0)
        eq(self.ioctx.read("abc"), b"rzxrzxrzx")
        [i.remove() for i in self.ioctx.list_objects()]

if __name__ == '__main__':
```
test results:
```
[root@jiaowa build]# python test_writesame_api.py 
...
----------------------------------------------------------------------
Ran 3 tests in 8.762s

OK

```